src/components/page-step/index.tsx

```diff
@@ -0,0 +1,279 @@
+import { Typography } from '@equinor/eds-core-react';
+import * as PropTypes from 'prop-types';
+import { Component } from 'react';
+import { mapRouteParamsToProps } from '../../utils/routing';
+import { connect } from 'react-redux';
+
+import { Breadcrumb } from '../breadcrumb';
+import { DocumentTitle } from '../document-title';
+import { routes } from '../../routes';
+import {
+  subscribeJob,
+  subscribeJobLogs,
+  subscribePipelineRuns,
+  unsubscribeJob,
+  unsubscribeJobLogs,
+  unsubscribePipelineRuns,
+} from '../../state/subscriptions/action-creators';
+import { routeWithParams, smallJobName } from '../../utils/string';
+
+import './style.css';
+import { Dispatch } from 'redux';
+import { RootState } from '../../init/store';
+import AsyncResource from '../async-resource';
+import { ScanOutput } from './scan-output';
+import { ScanStatus } from '../../models/scan-status';
+import { Code } from '../code';
+import { StepModel, StepModelValidationMap } from '../../models/step';
+import { PipelineRuns } from '../pipeline-runs';
+import { getPipelineRuns } from '../../state/pipeline-runs';
+import {
+  PipelineRunModel,
+  PipelineRunModelValidationMap,
+} from '../../models/pipeline-run';
+import {
+  getPipelineStepDescription,
+  getPipelineStepTitle,
+} from '../../utils/pipeline';
+import RelativeToNow from '../time/relative-to-now';
+import Duration from '../time/duration';
+import { getExecutionState } from '../component/execution-state';
+import { getStep } from '../../state/job';
+import { getJobStepLog } from '../../state/job-logs';
+
+const isStepRunning = (step: any) => step && !step.ended && step.started;
+
+export interface PagePipelineStepsSubscription {
+  subscribe: (props: string, string) => void;
+  unsubscribe: (props: string, string) => void;
+}
+
+export interface PageStepsProps extends PagePipelineStepsSubscription {
+  appName: string;
+  jobName: string;
+  step: StepModel;
+  pipelineRuns: Array<PipelineRunModel>;
```

Comment:
`step` and `pipelineRuns` are not marked with isRequired in propTypes declaration, but are marked mandatory in this interface.

looking at how these variables are used in your code suggests they are intended to be optional

```suggestion
  step?: StepModel;
  pipelineRuns?: Array<PipelineRunModel>;
```